telemetry/otelgrpc/otelgrpc.go

```diff
@@ -0,0 +1,210 @@
+package otelgrpc
+
+import (
+	"context"
+	"net"
+	"strings"
+	"time"
+
+	"github.com/goto/salt/utils"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+	semconv "go.opentelemetry.io/otel/semconv/v1.20.0"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/peer"
+	"google.golang.org/protobuf/proto"
+)
+
+type UnaryParams struct {
+	Start  time.Time
+	Method string
+	Req    any
+	Res    any
+	Err    error
+}
+
+type Meter struct {
+	duration     metric.Int64Histogram
+	requestSize  metric.Int64Histogram
+	responseSize metric.Int64Histogram
+	attributes   []attribute.KeyValue
+}
+
+type MeterOpts struct {
+	meterName string
+}
+
+func WithMeterName(meterName string) MeterOpts {
+	return MeterOpts{
+		meterName: meterName,
+	}
+}
+
+func NewMeter(hostName string, meterOpts ...MeterOpts) Meter {
+	meter := otel.Meter("github.com/goto/salt/telemetry/otelgrpc")
```

Comment:
The name should not be fixed like this. The one that uses library should be able to configure this